### PR TITLE
Feature/cypress testing

### LINKED ISSUE #83: Testing: CurrentTrip User Flow Sad
- User enters Trip Info
- User Starts Trip
- User sees ETA modal display
- User can select 'Start Trip'
-User can see a countdown timer
-User does not End Trip and sees option to add more time
-User does not add more time and sees an alert that a message has been sent to their contact that their trip was not completed

### LINKED ISSUE #85: Testing - CurrentTrip User Flow Happy Path
- User Creates Trip
- User selects 'Submit Trip'
- User sees modal with ETA and 'start trip' button
- User selects 'start trip' button
- User sees new view with countdown timer
-User ends trip before countdown timer expires and is redirected to the home page

### LINKED ISSUE #88: Testing - Add Contact Happy Paths
-User can navigate to Add Contact page from main view
-User enters information correctly
-User sees alert for contact added successfully

### PR DESCRIPTION
**Pull request by carolineshowscode**
## Type of Change
- [ ] JSX
- [ ] HTML
- [ ] JavaScript
- [ ] CSS
- [x] React
- [ ] React Router
- [x] Cypress
- [ ] Apollo
- [ ] GraphQL
- [ ] Twilio API
- [x] Refactor/Fix Bugs

### Fix:
- correct `Form.js` input testing for Autocomplete fields
- correct `Form.js` `Select` components so they can be selected and tested
- issues trying to work around testing google Places API autocomplete fields - resolved
- issues trying to work around testing Select components - resolved

### Feature:
- Prevented 'Enter' button from prematurely submitting `Form` 
- `Main Page` view and `Form` fields tested
- `Nav` view tested
- `AddContact` view tested
- `MainView` user flow tested
      - happy path: user starts and completes trip
      - sad path: user starts trip and does not select 'end trip'
-`AddContact` user flow tested
    -happy path: user successfully adds a new contact

   

## How has this been tested?
- [ ] In Browser
- [ ] In Console
- [x] Cypress
- [ ] Postman

### CODE REVIEW

File: cypress/integration/currentTrip-user-flow.js [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+    })
+  })
+
+  describe('Should be able to start a new trip', () => {
+
+    it('Should be able to complete form and start a new trip', () => {
+      cy.visit('http://localhost:3000/')
+      cy.get('.start-point').type('2343 Quitman St Denver')
+          .wait(200)
+          .type('{downarrow}')
```
Comment (pcmueller) [inline cypress/integration/currentTrip-user-flow.js:21]: Really cool solution, nice work figuring out the .type syntax to click/select different elements.

File: src/Components/Form/Form.js [hunk 2]
```diff
@@ -119,6 +124,7 @@ function Form({ contacts, handleEtaChange, userInfo, setContact, setTripIsActive
         value={travelMode}
         defaultValue={travelMode}
         onChange={setTravelMode}
+        onKeyPress={(e) => { e.key === 'Enter' && e.preventDefault(); }}
```
Comment (pcmueller) [inline src/Components/Form/Form.js:127]: Good call adding the conditional rendering to prevent a default key event!

**Review by pcmueller — APPROVED** [review]
Amazing work on all of this, thanks so much for taking on this huge task!

File: cypress/integration/addcontact.js [hunk 3]
```diff
@@ -1,31 +1,66 @@
...
+      cy.get('#react-burger-menu-btn')
+        .click()
+        .get('.menu-item').eq(1)
+          .click()
+    });
+
+    it('Should display an add contact form', () => {
+      cy.get('.add-contact').should('be.visible')
+        .get('h2').contains('Add Contact')
+        .get('h3').contains('Phone Number')
```
Comment (bhohnco) [inline cypress/integration/addcontact.js:30]: Nice work with AddContacts testing flow!

File: cypress/integration/addcontact.js [hunk 4]
```diff
@@ -1,31 +1,66 @@
...
+      .get('input[title="lastName"]').type('Test')
+      .get('input[title="countryCode"]').type(1)
+      .get('input[title="areaCode"]').type(720)
+      .get('input[title="phoneNumber"]').type(1234567)
+      .get('.add-contact-btn').click()
+    });
+
+    it('Shows the new contact when you visit the new trip form', () => {
+      cy.get('#react-burger-menu-btn').click()
+        .get('.menu-item').eq(0)
```
Comment (bhohnco) [inline cypress/integration/addcontact.js:59]: Nice, I wouldn't have thought of this

File: cypress/integration/currentTrip-user-flow.js (hunk 1 above)
Comment (bhohnco) [reply cypress/integration/currentTrip-user-flow.js:21]: ^^^ I agree. The down arrow is a great addition

**Pull request merged**